types/react-bootstrap-table/index.d.ts

```diff
@@ -1,66 +1,207 @@
-// Type definitions for react-bootstrap-table 2.6
+// Type definitions for react-bootstrap-table 4.0
 // Project: https://github.com/AllenFang/react-bootstrap-table
-// Definitions by: Frank Laub <https://github.com/flaub>, Aleksander Lode <https://github.com/alelode>, Josué Us <https://github.com/UJosue10>
+// Definitions by: Frank Laub <https://github.com/flaub>,
+//                 Aleksander Lode <https://github.com/alelode>,
+//                 Josué Us <https://github.com/UJosue10>
+//                 Janeene Beeforth <https://github.com/dawnmist>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.3
 
-/// <reference types="node" />
-
 // documentation taken from http://allenfang.github.io/react-bootstrap-table/docs.html
 
-import { ComponentClass, Props, ReactElement } from 'react';
-import { EventEmitter } from 'events';
+import { Component, CSSProperties, Props, ReactElement, SyntheticEvent } from 'react';
+
+/**
+ * Row Keys type. Used as the unique primary index for a row.
+ * This should be either a string or a number.
+ */
+export type RowKey<T extends number | string> = T;
+
+/**
+ * Cell value type.
+ */
+export type Cell<T extends string | boolean | number | Date | object | string[] | boolean[] | number[] | Date[] | object[]> = T;
```

Comment:
As with `RowKey` above, this is only ever used as `Cell<any>` and as such is effectively just `any`.